### ISSUE TITLE
No additional power used when changing the action queue of a robot

### ISSUE THREAD

**Issue by FlorianSchroeter** [Lux-AI-Challenge/Lux-Design-2022#112]
According to the documentation regarding changes to the action queue:

> Submitting a new action queue for a robot requires the robot to use additional power to replace it's action queue. It costs an additional 1 power for Lights, an additional 10 power for Heavies.

https://www.lux-ai.org/specs-2022-beta#Actions

I would assume that changing the action queue of a heavy robot every turn to move in a different direction onto a rubble free tile would cost 20 (move) + 10 (action queue change) power. Yet I cannot confirm this behaviour locally. Even changing the action queue from a move action to a dig action does only ever reduce the power by the amount required for the action itself.

Am I missing something? Or could this be a bug in the implementation of Lux Eye 2022?

This is how I am changing the action queue right now. It is by no means designed to be a final implementation but to elaborate on my understanding of the rule set:
>
        if units.items():
            self.direction += 1
            if self.direction > 4:
                self.direction = 0
        for unit_id, unit in units.items():
            # Test power consumption for changing the action queue
            if self.direction == 0:
                actions[unit_id] = [unit.dig(repeat=False)]
            elif self.direction < 0:
                actions[unit_id] = [unit.move(4, repeat=False)]
            else:
                actions[unit_id] = [unit.move(self.direction, repeat=False)]
            continue

self.direction is initialized with a negative value, so that the units will move to the left (4) and out of the factory first.

**Comment by StoneT2000** [comment]
Taking a look

**Comment by StoneT2000** [comment]
Not sure what might be happening but perhaps the additional 10 power the heavy gets is added in and that's why it looks like it doesn't take extra 10 power.

**Comment by FlorianSchroeter** [comment]
Yes, I can confirm your assumption. If I modify my code to change the action queue only once assigning a move action in the process, the heavy unit looses 20 energy on the first turn (+ 10 solar power -10 action queue change - 20 move onto rubble free tile = 20) and 10 energy every later turn (+ 10 solar power - 20 move onto rubble free tile) .

So a unit regains power every turn (during daytime) regardless of the performed action. Thank you very much for the clarification. I just checked the documentation and I guess I missed this part:

> During the day, solar panels replenish the power of all robots but during the night robots power is not recharged.

Closing this issue now - sorry for the inconvenience.

**Comment by FlorianSchroeter** [comment]
Yes, I can confirm your assumption. If I modify my code to change the action queue only once assigning a move action in the process, the heavy unit looses 20 energy on the first turn (+ 10 solar power -10 action queue change - 20 move onto rubble free tile = 20) and 10 energy every later turn (+ 10 solar power - 20 move onto rubble free tile) .

So a unit regains power every turn (during daytime) regardless of the performed action. Thank you very much for the clarification. I just checked the documentation and I guess I missed this part:

> During the day, solar panels replenish the power of all robots but during the night robots power is not recharged.

Closing this issue now - sorry for the inconvenience.
